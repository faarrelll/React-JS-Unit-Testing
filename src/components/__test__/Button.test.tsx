import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { describe, expect, test } from "vitest";
import { Button } from "../ui/button";

describe("Reusable Button component usage", () => {
  test("should render the button", () => {
    const { asFragment } = render(<Button>Click me</Button>);
    expect(asFragment()).toHaveTextContent("Click me");
    expect(asFragment()).toMatchSnapshot();
  });

  test("should render the button with a onClick function state", () => {
    const onClick = () => {
      console.log("clicked");
    };
    const { asFragment } = render(<Button onClick={onClick}>Click me</Button>);
    expect(asFragment()).toHaveTextContent("Click me");
    expect(asFragment()).toMatchSnapshot();
  });
});
