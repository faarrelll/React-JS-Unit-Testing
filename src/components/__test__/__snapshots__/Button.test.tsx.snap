// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Reusable Button component usage > should render the button 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;

exports[`Reusable Button component usage > should render the button with a onClick function state 1`] = `
<DocumentFragment>
  <button
    class="inline-flex items-center justify-center gap-2 whitespace-nowrap rounded-md text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg]:size-4 [&_svg]:shrink-0 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-9 px-4 py-2"
  >
    Click me
  </button>
</DocumentFragment>
`;
